gender_analysis/analysis/corpus_metadata_visualizations.py

```diff
@@ -0,0 +1,209 @@
+import matplotlib.pyplot as plt
+import seaborn as sns
+from collections import Counter
+
+from gender_analysis.common import MissingMetadataError
+
+
+def plt_pubyears(corpus):
+    """
+    Creates a histogram displaying the frequency of books that were published within a 20 year 
+    period
+    Requires that corpus contains a 'date' metadata field
+    :param corpus: Corpus
+    RETURNS a pyplot histogram
+    """
+    if 'date' not in corpus.get_corpus_metadata():
+        raise MissingMetadataError('This corpus does not contain metadata on publication date.')
+
+    pub_years = []
+    for doc in corpus.documents:
+        if doc.date is None:
+            continue
+        pub_years.append(doc.date)
+
+    if corpus.name:
+        corpus_name = corpus.name
+    else:
+        corpus_name = 'corpus'
+
+    sns.set_style('ticks')
+    sns.color_palette('colorblind')
+    ax1 = plt.subplot2grid((1, 1), (0, 0))
+    plt.figure(figsize=(10, 6))
+    bins = [num for num in range(min(pub_years), max(pub_years)+4, 5)]
+    plt.hist(pub_years, bins, histtype='bar', rwidth=.8, color='c')
+    plt.xlabel('Year', size=15, weight='bold', color='k')
+    plt.ylabel('Frequency', size=15, weight='bold', color='k')
+    plt.title('Publication Year Concentration for '+corpus_name.title(), size=18, weight='bold',
+              color='k')
+    plt.yticks(size=15, color='k')
+    plt.xticks([i for i in range(min(pub_years), max(pub_years)+9, 10)], size=15, color='k')
+    for label in ax1.xaxis.get_ticklabels():
+        label.set_rotation(60)
+    plt.subplots_adjust(left=.1, bottom=.18, right=.95, top=.9)
+    plt.savefig('date_of_pub_for_'+corpus_name.replace(' ', '_')+'.png')
+
+
+def plt_pubcountries(corpus):
+    """
+    Creates a bar graph displaying the frequency of books that were published in each country
+    Requires that corpus contains a 'country_publication' metadata field
+    :param corpus: Corpus
+    RETURNS a pyplot bargraph
+    """
+    if 'country_publication' not in corpus.get_corpus_metadata():
+        raise MissingMetadataError('This corpus does not contain metadata on publication country.')
```

Comment:
Same comment as above -- we don't explicitly explain what's missing in this exception.